apps/alert_processor/lib/model/query.ex

```diff
@@ -183,6 +183,57 @@ defmodule AlertProcessor.Model.Query do
         LEFT JOIN users_before_end ON users_before_end.period_end = periods.end
         ORDER BY periods.end DESC
         """
+      },
+      %__MODULE__{
+        label: "Grouped Time Windows",
+        query: """
+        SELECT
+          to_char(s.start_time, 'FMHH24:MI') as start_time,
+          to_char(s.end_time, 'FMHH24:MI') as end_time, s.type,
+          count(distinct s.id)
+        FROM subscriptions s
+        GROUP BY s.type, s.start_time, s.end_time;
+        """
+      },
+      %__MODULE__{
+        label: "Fully Paused Accounts",
+        query: """
+          WITH
```

Comment:
Please let me know if there's some other obvious way to do this one.

Reply:
Nice! With the caveat that I'm far from a SQL master, I'd say that this all seems reasonable to me. And more than that, it's really incredibly readable for what it is, which I appreciate.

Reply:
Yeah I am also not an SQL master 😅 The unions feel a bit weird but if it works it works (I tested it with several accounts and subscriptions, seems solid enough). Thanks!